flag.actions.inc

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * @file
+ * Action callbacks for flag module.
+ */
+
+/**
+ * Implements Drupal action. Flags any entity.
+ */
+function flag_entity_action(&$entity, $context = array()) {
+  $entity_flags = flag_get_flags($context['action_info']['type']);
+  if (count($entity_flags) === 1) {
+    $flag = reset($entity_flags);
+    $account = isset($context['account']) ? $context['account'] : $GLOBALS['user'];
+    $op = ($flag->is_flagged($entity->id()))? 'unflag' : 'flag';
+    $flag->flag($op, $entity->id(), $account, TRUE);
+  }
+  else {
+    $entity_info = entity_get_info($context['action_info']['type']);
+    $message = 'This action can only flag or unflag items from this interface if there is only a single flag for @label';
+    $substitutions = array('@label' => $entity_info['label']);
+    backdrop_set_message(t($message, $substitutions), 'warning');
+    return FALSE;
+  }
+}
+
+/**
+ * Form for configuring the Flag node action.
+ */
+function flag_node_action_form($context = array()) {
```

Comment:
I'm not sure if these get called. From what I recall, core version of Bulk Actions didn't include the ability to have forms for configuring actions.